### PR TITLE
fix: wallet crash on web3-id signin

### PR DESCRIPTION
**Pull request by maycon-mello**
https://dock-team.atlassian.net/browse/DCKA-1197

The error handler was trying to render an object as a react element, which crashes react-native.  This might be related to some API change

### CODE REVIEW

File: src/features/qr-code-scanner/qr-code.js [hunk 1]
```diff
@@ -153,6 +153,25 @@ export function onAuthQRScanned(data) {
...
 }
+
+export function getWeb3IdErrorMessage(result) {
+  let error;
+  try {
+    const apiError = result.error.results[0].error;
+
+    if (typeof apiError !== 'string') {
+      throw new Error(
+        `Error is not a string, received: ${JSON.stringify(apiError)}`,
```
[the remark this answers is not in the document]
Comment (maycon-mello) [reply src/features/qr-code-scanner/qr-code.js:164]: I can remove this check, anyway the error will be sent to sentry and the user will see a default message `locale.auth.auth_sign_in_failed`

[the remark this answers is not in the document]
Comment (maycon-mello) [reply src/features/qr-code-scanner/qr-code.js:164]: in the current api response the error will be received on `result.error.results[0].error`

[the remark this answers is not in the document]
Comment (maycon-mello) [reply src/features/qr-code-scanner/qr-code.js:164]: was able to grab this one while debuging:
```{
  "verified": false,
  "error": {
    "verified": false,
    "results": [
      {
        "proof": {
          "@context": [
            "https://www.w3.org/2018/credentials/v1",
            {
              "dk": "https://ld.dock.io/credentials#",
              "DockAuthCredential": "dk:DockAuthCredential",
              "name": "dk:name",
              "email": "dk:email",
              "state": "dk:state",
              "IssuerPolicy": "dk:IssuerPolicy",
              "AllVerifiers": "dk:AllVerifiers",
              "Archival": "dk:Archival",
              "prohibition": "dk:prohibition",
              "action": "dk:action",
              "assignee": "dk:assignee",
              "assigner": "dk:assigner",
              "target": "dk:target"
            }
          ],
          "type": "Sr25519Signature2020",
          "created": "2022-11-24T17:23:15Z",
          "verificationMethod": "did:dock:5CNunNiQbFXnf5bC555nGzqfgcEA7rgQprhLYc7Yz5pVkA42#5H4MZpxujyZErMseJ87oDVn8eFKaCozporcP4esEp39DrkkV",
          "proofPurpose": "assertionMethod",
          "proofValue": "z346Dv5frAx8nRJsfeUSreTrHZrk25QDdjV3oHFJPsDAciL8ghSxcxwxUjiBQXNJiHePVF7VjnJxd2NuQRwBz7w8R"
        },
        "verified": false,
        "error": "verification method should have type Sr25519VerificationKey2020 - got: undefined"
      }
    ]
  },
  "userId": "did:dock:5CNunNiQbFXnf5bC555nGzqfgcEA7rgQprhLYc7Yz5pVkA42"
}`

[the remark this answers is not in the document]
Comment (SamHellawell) [reply src/features/qr-code-scanner/qr-code.js:164]: we will fix up these errors eventually so it's not as complicated to access, requires some sdk fixes

**Review by SamHellawell — APPROVED** [review]

File: src/features/qr-code-scanner/qr-code.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (maycon-mello) [reply src/features/qr-code-scanner/qr-code.js:164]: sure, thanks

**Pull request merged**